docs/spec/parameters.md

```diff
@@ -13,95 +13,99 @@ param myObject object
 param myArray array
 ```
 
-A parameter cannot have the same name as a [variable](./variables.md), [resource](./resources.md), [output](./outputs.md) or another parameter in the same scope.
-
-## Secure parameters
-If you are familiar with ARM template parameters, you will notice a conspicuous absense of `secureString` and `secureObject` types. In this language, these types are implemented as modifiers.
-
-The following declarations will compile into a `secureString` and `secureObject` parameters, respectively.
+## Default value
+Default values can be declared as follows:
 ```
-param myPassword string { 
-  secure: true
-}
-
-param mySuperSecretObject object { 
-  secure: true
-}
+param myParam string = 'my default value'
 ```
-## Allowed Values
-You can constrain which values are allowed using the `allowed` modifier:
+
+You may use [expressions](./expressions.md) with the `default` modifier.  Here is an example of a location parameter whose value defaults to the location of the current resource group if the parameter is not specified during the deployment:
 ```
-param myEnum string {
-  allowed: [
-    'one'
-    'two'
-  ]
-}
+param myParam string = resourceGroup().location
 ```
 
-The constraint will be evaluated at deployment time of the compiled template.
+A parameter cannot have the same name as a [variable](./variables.md), [resource](./resources.md), [output](./outputs.md) or another parameter in the same scope.
+
+## Declration with decorators
+Decorators provide a way to attach constrains and metadata to a parameter. Decorators are placed above the paramter declaration to be decorated. They use the form @expression, where expression must be a function call:
```

Comment:
>paramter [](start = 107, length = 8)

typo

Reply:
Fixed. I just found this VSCode [extension](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) and installed it. Not surprisingly it discovered a bunch more typos in the code and docs...I will send a separate PR to fix all of them.

Reply:
Wow 2M+ downloads! Just installed it too!

---
In reply to: [588794961](https://github.com/Azure/bicep/pull/1693#discussion_r588794961) [](ancestors = 588794961)